// ng-semantic/ng-semantic/popup/popup.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var SMPopupDirective = (function () {
    function SMPopupDirective(element) {
        this.element = element;
    }
    SMPopupDirective.prototype.onClick = function () {
        if (typeof jQuery === "undefined") {
            console.log("jQuery is not loaded");
            return;
        }
        if (!this.smDirPopup.hasOwnProperty("position") || !this.smDirPopup.hasOwnProperty("selector")) {
            console.log("position or target selector missing for popup");
            return;
        }
        jQuery(this.element.nativeElement).popup({
            closable: true,
            exclusive: true,
            lastResort: true,
            on: "click",
            popup: "." + this.smDirPopup.selector + ".popup",
            position: this.smDirPopup.position,
            preserve: true
        }).popup("show");
    };
    __decorate([
        core_1.Input(), 
        __metadata('design:type', Object)
    ], SMPopupDirective.prototype, "smDirPopup", void 0);
    __decorate([
        core_1.HostListener("click", ["$event.target"]), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', []), 
        __metadata('design:returntype', void 0)
    ], SMPopupDirective.prototype, "onClick", null);
    SMPopupDirective = __decorate([
        core_1.Directive({
            selector: "[smDirPopup]"
        }), 
        __metadata('design:paramtypes', [core_1.ElementRef])
    ], SMPopupDirective);
    return SMPopupDirective;
}());
exports.SMPopupDirective = SMPopupDirective;
var SemanticPopupComponent = (function () {
    function SemanticPopupComponent() {
    }
    __decorate([
        core_1.Input("selector"), 
        __metadata('design:type', String)
    ], SemanticPopupComponent.prototype, "selector", void 0);
    SemanticPopupComponent = __decorate([
        core_1.Component({
            selector: "sm-popup",
            template: "<div class=\"ui popup very wide {{selector}}\">\n    <div class=\"content\">\n        <ng-content></ng-content>\n    </div>\n</div>"
        }), 
        __metadata('design:paramtypes', [])
    ], SemanticPopupComponent);
    return SemanticPopupComponent;
}());
exports.SemanticPopupComponent = SemanticPopupComponent;
